### PR TITLE
Update Cypress Best Practices with test retry documentation

### PR DESCRIPTION
**Pull request by ddzz**

### CODE REVIEW

File: platform/testing/end-to-end/cypress-best-practices-on-vsp.md [hunk 1]
```diff
@@ -71,6 +72,14 @@ Cypress queues its commands instead of running them synchronously, so doing some
 
 Cypress has many third party [plugins](https://docs.cypress.io/plugins/) available. If you find yourself needing to do something that isn't natively supported, there may be a plugin for it.
 
+### Test retries
+
+When running tests on Jenkins, CircleCI, or locally in headless mode, Cypress will retry failing tests up to two times before marking the test as failed. This is meant to reduce the number of build failures from tests that fail occasionally.
+
+Tests should be written to be retry-able in order to use this Cypress feature effectively. Without proper test structure and implementation (set up, execution, teardown, etc.), tests can fail consistently upon retry.
```
Comment (1Copenut) [inline platform/testing/end-to-end/cypress-best-practices-on-vsp.md:79]: Is there anything specific about how tests should be written to be retry-able? I'm approaching this as someone completely new to our tech stack (or at least Cypress). I would be looking for a link to a separate section or a small code snippet to help me write my test in the intended manner.

**Review by 1Copenut — COMMENTED** [review]
@ddzz I'm submitting my review as a general comment only. I like the way this is structured, and asked a review question if there's a way to inform users how to write good tests. Maybe that's baked into Cypress and no further action is needed, in that case I have no changes and it LGTM.

File: platform/testing/end-to-end/cypress-best-practices-on-vsp.md (hunk 1 above)
Comment (ddzz) [reply platform/testing/end-to-end/cypress-best-practices-on-vsp.md:79]: Thanks for the comment, I wasn't sure how much detail to go into in this document. I've updated the PR with a simple Google search test example.

Comment (1Copenut) [reply platform/testing/end-to-end/cypress-best-practices-on-vsp.md:79]: Thank you @ddzz . This has such an immediate impact, reading through the what and seeing the how right after.

**Review by 1Copenut — APPROVED** [review]
👍 LGTM!

**Review by pjhill — APPROVED** [review]
LGTM

**Pull request merged**